### PR TITLE
pin fulcio deps

### PR DESCRIPTION
**Pull request by marcofranssen**
- Allow committing Chart.lock, exclude charts/*.tgz
- Pin fulcio dependency versions
- Add fulcio README.md.gotmpl and generate docs

 **Description of the change**

 **Existing or Associated Issue(s)**

#298 

 **Additional Information**

 

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). Where applicable, update and bump the versions in any associated umbrella chart
- [x] Variables are documented in the `values.yaml` and added to the README.md. THe [helm-docs](https://github.com/norwoodj/helm-docs) utility can be used to generate the necessary content. Use `helm-docs --dry-run` to preview the content
- [x] JSON Schema generated
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command

### CODE REVIEW

File: charts/fulcio/README.md.gotmpl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+> server:
+>   ingress:
+>     http:
+>       enabled: true
+>       hosts:
+>         - host: fulcio.localhost
+>           path: /
+>     grpc:
+>       enabled: false
+> ```
```
Comment (cpanato) [inline charts/fulcio/README.md.gotmpl:77]: nit: extra line

File: ct.yaml [hunk 2]
```diff
@@ -3,3 +3,6 @@ chart-dirs:
 validate-maintainers: false
 remote: origin
 target-branch: main
+chart-repos:
+  - sigstore=https://sigstore.github.io/helm-charts
+
```
Comment (cpanato) [inline ct.yaml:8]: nit: extra line

File: charts/fulcio/README.md [hunk 3]
```diff
@@ -90,40 +76,44 @@ The following table lists the configurable parameters of the Fulcio chart and th
...
-| server.ingress.grpc.className | object | `""` |  |
-| server.ingress.grpc.enabled | bool | `true` |  |
-| server.ingress.grpc.hosts | array | [] |  |
-| server.ingress.grpc.hosts.host | string |  |  |
-| server.ingress.grpc.hosts.path | string | `"/"` |  |
-| server.ingress.grpc.hosts.service_name | string |  |  |
-| server.ingress.grpc.tls | array | `[]` |  |
-| server.ingress.grpc.tls.hosts | array | `[]` |  |
-| server.ingress.grpc.tls.secretName | string | `` |  |
+| server.image.version | string | `"sha256:61081295a8f75ed7537b5d1f8c7320e078dc00e4562c0bf605fbefa062c690de"` |  |
```
Comment (cpanato) [inline charts/fulcio/README.md:87]: can you add the tag version here? thanks

File: charts/fulcio/Chart.lock [hunk 4]
```diff
@@ -0,0 +1,6 @@
+dependencies:
```
Comment (cpanato) [inline charts/fulcio/Chart.lock:1]: i'm good to have this

@sabre1041 does that is ok for you?

**Review by cpanato — CHANGES_REQUESTED** [review]
few comments

thanks for working on this

File: charts/fulcio/Chart.lock (hunk 4 above)
Comment (marcofranssen) [reply charts/fulcio/Chart.lock:1]: Probably good to mention is that it is easy to manage this see #300 on how folks can update when bumping the dependency versions.

**Comment by marcofranssen** [comment]
Handled the feedback. Thanks for catching those.

File: charts/fulcio/values.yaml [hunk 5]
```diff
@@ -12,8 +12,8 @@ server:
     registry: gcr.io
     repository: projectsigstore/fulcio
     pullPolicy: IfNotPresent
-    # crane digest gcr.io/projectsigstore/fulcio:v0.5.3
-    version: "sha256:61081295a8f75ed7537b5d1f8c7320e078dc00e4562c0bf605fbefa062c690de"
+    # -- v0.5.3
```
Comment (marcofranssen) [inline charts/fulcio/values.yaml:15]: The crane command is now documented in CONTRIBUTING.md see #300

Comment (priyawadhwa) [reply charts/fulcio/values.yaml]: I would strongly prefer to leave this comment in the file as well, I don't think it's likely someone doing a release will read CONTRIBUTING.md before starting

**Review by priyawadhwa — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Pull request merged**